File: packages/react-native-performance/android/src/main/kotlin/com/shopify/reactnativeperformance/Interactive.kt
package com.shopify.reactnativeperformance

// On iOS, we cannot send back non-object types to JS via a dictionary,
// so we have to use strings. Legal values: `TRUE` and `FALSE`.
// Using the same convention of android for consistency.
private const val TRUE_STRING = "TRUE"
private const val FALSE_STRING = "FALSE"

class Interactive(stringValue: String) {
  val boolValue: Boolean

  init {
    if (stringValue == TRUE_STRING) {
      this.boolValue = true
    } else if (stringValue == FALSE_STRING) {
      this.boolValue = false
    } else {
      throw IllegalArgumentException("The only allowed values for the 'interactive' prop are '$TRUE_STRING' and '$FALSE_STRING', but it was '$stringValue'.")
    }
  }

  override fun toString() = if (boolValue) TRUE_STRING else FALSE_STRING
}

File: packages/react-native-performance/ios/ReactNativePerformance/Interactive.swift
import Foundation

final class Interactive {
    private static let TRUE_STRING = "TRUE"
    private static let FALSE_STRING = "FALSE"

    let boolValue: Bool

    init?(stringValue: String) {
        switch stringValue {
        case Interactive.TRUE_STRING:
            boolValue = true
        case Interactive.FALSE_STRING:
            boolValue = false
        default:
            assertionFailure("The only allowed values for the 'interactive' prop are '\(Interactive.TRUE_STRING)' and '\(Interactive.FALSE_STRING)', but it was '\(stringValue)'.")
            return nil
        }
    }

    public var description: String {
        return boolValue ? Interactive.TRUE_STRING : Interactive.FALSE_STRING
    }
}
